src/gerrit/review/review.py

```diff
@@ -23,24 +24,32 @@ def main():
     get_env('INFRABOX_GERRIT_USERNAME')
     get_env('INFRABOX_GERRIT_KEY_FILENAME')
 
+    cluster_name = get_env('INFRABOX_CLUSTER_NAME')
+
     conn = connect_db()
     conn.set_isolation_level(psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT)
     logger.info("Connected to database")
 
-    elect_leader(conn, "gerrit-review")
+    elect_leader(conn, 'gerrit-review' ,cluster_name)
 
     curs = conn.cursor()
     curs.execute("LISTEN job_update;")
 
     logger.info("Waiting for job updates")
 
     while True:
-        is_leader(conn, "gerrit-review")
-        if select.select([conn], [], [], 5) != ([], [], []):
-            conn.poll()
-            while conn.notifies:
-                notify = conn.notifies.pop(0)
-                handle_job_update(conn, json.loads(notify.payload))
+        if not is_active(conn, cluster_name):
```

Comment:
have you checked if leader election works properly? I think there was an issue with it.

Reply:
I have checked and tested.

1. At the first, clusters competes for leader in transaction, so only one cluster wins.

2. The elected leader will refresh last_activity_time repeatedly, keeping himself the leader.

3. If the leader fails for more than 30 seconds, other clusters will compete for leader